gc.c

```diff
@@ -9789,7 +9789,7 @@ compat_key(VALUE key)
     if (!NIL_P(new_key)) {
 	static int warned = 0;
 	if (warned == 0) {
-	    rb_warn("GC.stat keys were changed from Ruby 2.1. "
+            rb_category_warn(RB_WARN_CATEGORY_DEPRECATED, "GC.stat keys were changed from Ruby 2.1. "
```

Comment:
In general, I'm in favoring of deprecating in version X.Y and removing in version X.Y+1.  Some of the deprecation warnings have been in place far longer than 1 one version.  However, since Ruby 3.0 release is in one week, I thought it best to be conservative.  We can always remove the deprecated feature after 3.0 release.